✨ feat: 회원 관리 페이지 조회 API 개발

Syapa723 opened the pull request:
회원 관리 페이지 조회 기능 serializer, view, url 작성
django-filter-stubs 의존성 패키지 설치

## ✅ PR 요약
- 관련 이슈 번호: #105 
- 작업 요약: 회원 관리 페이지 회원 목록 조회 기능 구현

## 📄 상세 내용
- [x] 회원 목록 조회 기능 구현을 위한 serializer, view, url 작성
- [x] view의 코드 가독성과 추후 개발예정인 API (대시보드)에서 다시 활용이 가능한 filters.py 작성
- [x] mypy 에러 해결을 위한 django-filter-stubs 의존성 패키지 설치


## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Syapa723 committed:
✨ feat: 회원 관리 페이지 조회 API 개발

회원 관리 페이지 조회 기능 serializer, view, url 작성
django-filter-stubs 의존성 패키지 설치

Syapa723 committed:
🚚 build: poetry conflict 해결

SimJunSik commented on apps/users/filters.py:
```diff
@@ -0,0 +1,47 @@
+
+    def filter_by_permission(self, queryset: QuerySet[User], name: str, value: str) -> QuerySet[User]:
+        # 파라미터 값에 따라 is_superuser와 is_staff 값을 조합해서 필터링
+        if value == "admin":
```
admin, staff, general 문자열 들을 enum 으로 묶으면 좋을 것 같습니다
아래 withdrawn 같은 것들도!

Meoyoug replied:
```
PERMISSION_CHOICES = (
    ("admin", "관리자"),
    ("staff", "스태프"),
    ("general", "일반회원"),
)
STATUS_CHOICES = (
    ("active", "활성화"),
    ("inactive", "비활성화"),
    ("withdrawn", "탈퇴진행중"),
)
```
상기에 정의한 이것들 enums.py에 Enum상속받아서 구현하면 활용도가 높을 것 같아요!
나중에 유지보수하기도 쉽구요

Meoyoug commented on apps/users/tests/test_admin.py:
```diff
@@ -45,29 +45,29 @@ def setUp(self) -> None:
-        #     user=self.withdrawn_user, reason="OTHER", reason_detail="test", due_date="2025-12-31"
-        # )
-        # self.list_url = reverse("admin_user:users-list")
+        self.inactive_user = user_model.objects.create_user(
```
setUp 클래스의 경우 test 메서드가 실행될 때마다 호출됩니다!
따라서 모든 테스트 메서드에서 공통적으로 사용되는 데이터들에 한해서 정의해주는게 좋으며
그 이외에 given절에 해당하는 데이터가 있다면 해당 메서드 내에서 정의해줘야만 테스트 효율을 증대시킬 수 있습니다

Meoyoug commented on apps/users/views/admin_views.py:
```diff
@@ -36,3 +41,31 @@ def patch(self, request: Request, user_uuid: UUID) -> Response:
+    max_page_size = 50
+
+
+class UserAdminViewSet(viewsets.ReadOnlyModelViewSet[User]):
```
ReadOnlyModelViewSet의 경우 해당 자원의 목록을 반환하는 ListAPIView 뿐만이 아니라 해당 자원의 pk로 접근하여 상세정보를 가져오는 RetrieveAPIView가 포함이 됩니다!
회원 상세 조회에 대한 테스트 코드가 추가되어야 할 것으로 보입니다!

Syapa723 committed:
✨ feat: 회원 목록 조회 및 상세 조회 구현

회원 목록 조회에서 권한과 회원상태 관련 문자열을 enums.py로 통합해 중앙에서 관리하도록 리팩토링, 상세 정보 조회를 가져오는 RetrieveAPIView 코드 추가

Syapa723 committed:
♻️ refactor: enums.py 코드 통합

users 디렉토리 enums.py 삭제 후 users/utils 디렉토리의 enums.py로 코드 통합, 코드 통합에 따른 import 구문 수정

Meoyoug reviewed (approved):
고생하셧씁니당
